docs/source/developers/cpp.rst

```diff
@@ -174,6 +169,12 @@ boolean flags to ``cmake``.
 * ``-DARROW_PARQUET=ON``: Apache Parquet libraries and Arrow integration
 * ``-DARROW_PLASMA=ON``: Plasma Shared Memory Object Store
 * ``-DARROW_PLASMA_JAVA_CLIENT=ON``: Build Java client for Plasma
+* ``-DARROW_WITH_BZ2=ON``: Build support for BZ2 compression
```

Comment:
Can we keep the list more or less alphabetical?

Reply:
done